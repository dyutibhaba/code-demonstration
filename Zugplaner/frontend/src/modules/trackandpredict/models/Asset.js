import PropTypes from 'prop-types';

export default class Asset {
  static empty() {
    return new Asset('', '', 0, '', '', 0.0, 0.0, 0, '', '', null, 0);
  }

  static fromBlob({
    name,
    batteryPercentage,
    batteryVoltage,
    lastLocationDateTime,
    positionAddress,
    latitude,
    longitude,
    speedKph,
    id,
    eta,
    etaVariation,
    damageStatusFlag,
  }) {
    return new Asset(
      name,
      batteryPercentage,
      batteryVoltage,
      lastLocationDateTime,
      positionAddress,
      latitude,
      longitude,
      speedKph,
      id,
      eta,
      etaVariation,
      damageStatusFlag,
    );
  }

  static fromRowData({
    name,
    batteryPercentage,
    batteryVoltage,
    lastLocationDateTime,
    positionAddress,
    latitude,
    longitude,
    speedKph,
    id,
    eta,
    etaVariation,
    damageStatusFlag,
  }) {
    return new Asset(
      name,
      batteryPercentage,
      batteryVoltage,
      lastLocationDateTime,
      positionAddress,
      latitude,
      longitude,
      speedKph,
      id,
      eta,
      etaVariation,
      damageStatusFlag,
    );
  }

  static get shape() {
    return PropTypes.shape({
      name: PropTypes.string,
      batteryPercentage: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
      batteryVoltage: PropTypes.number,
      lastLocationDateTime: PropTypes.string,
      positionAddress: PropTypes.string,
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      positionGPS: PropTypes.string,
      speedKph: PropTypes.number,
      id: PropTypes.string,
      eta: PropTypes.string,
      etaVariation: PropTypes.number,
      damageStatusFlag: PropTypes.number,
    });
  }

  constructor(
    name,
    batteryPercentage,
    batteryVoltage,
    lastLocationDateTime,
    positionAddress,
    latitude,
    longitude,
    speedKph,
    id,
    eta,
    etaVariation,
    damageStatusFlag,
  ) {
    this.name = name;
    this.batteryPercentage = batteryPercentage;
    this.batteryVoltage = batteryVoltage;
    this.lastLocationDateTime = lastLocationDateTime;
    this.positionAddress = positionAddress;
    this.latitude = latitude;
    this.longitude = longitude;
    this.positionGPS = `${latitude}, ${longitude}`;
    this.speedKph = speedKph;
    this.id = id;
    this.eta = eta;
    this.etaVariation = this.eta ? etaVariation : null;
    this.damageStatusFlag = damageStatusFlag;
  }
}
