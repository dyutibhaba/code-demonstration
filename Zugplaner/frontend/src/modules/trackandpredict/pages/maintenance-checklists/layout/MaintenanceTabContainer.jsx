import React from 'react';
import PropTypes from 'prop-types';
import MaintenanceLayout from './MaintenanceLayout';
import Asset from '../../../models/Asset';

const MaintenanceTabContainer = ({ children, asset, index }) => {
  const maintenanceLinks = [
    {
      to: () => `/track-and-predict/maintenance-asset/${asset.id}/checklists`,
      value: 'maintenance.tabs.checklists',
    },
    {
      to: () => `/track-and-predict/maintenance-asset/${asset.id}/damages`,
      value: 'maintenance.tabs.damages',
    },
    {
      to: () => `/track-and-predict/maintenance-asset/${asset.id}/maintenance-parameters`,
      value: 'maintenance.tabs.parameters',
    },
    {
      to: () => `/track-and-predict/maintenance-asset/${asset.id}/maintenance-log`,
      value: 'maintenance.tabs.regular',
    },
    {
      to: () => `/track-and-predict/maintenance-asset/${asset.id}/activity`,
      value: 'maintenance.tabs.activity',
    },
    {
      to: () => `/track-and-predict/maintenance-asset/${asset.id}/mileage`,
      value: 'maintenance.tabs.mileage',
    },
  ];

  return (
    <MaintenanceLayout
      links={maintenanceLinks.map((link) => ({
        ...link,
        to: link.to(),
      }))}
      breadcrumb={[
        {
          link: `/track-and-predict/maintenance-asset/${asset.id}/overview`,
          label: asset && asset.name ? asset.name : 'unknown',
        },
      ]}
      index={index}
    >
      {children}
    </MaintenanceLayout>
  );
};

MaintenanceTabContainer.defaultProps = {
  children: null,
};

MaintenanceTabContainer.propTypes = {
  children: PropTypes.node,
  asset: Asset.shape.isRequired,
  index: PropTypes.number.isRequired,
};

export default MaintenanceTabContainer;
