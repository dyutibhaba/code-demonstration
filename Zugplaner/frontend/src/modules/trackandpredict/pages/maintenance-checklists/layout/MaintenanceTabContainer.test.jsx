import React from 'react';
import renderWithState from '../../../../../testutils';
import Asset from '../../../models/Asset';
import MaintenanceTabContainer from './MaintenanceTabContainer';

describe('Maintenance Tab Container', () => {
  it('renders the maintenance tab container correctly', () => {
    expect(renderWithState({
      assetsWithState: [],
    })((
      <MaintenanceTabContainer asset={Asset.empty()} index={0}>
        Children
      </MaintenanceTabContainer>
    ))).toMatchSnapshot();
  });
});
