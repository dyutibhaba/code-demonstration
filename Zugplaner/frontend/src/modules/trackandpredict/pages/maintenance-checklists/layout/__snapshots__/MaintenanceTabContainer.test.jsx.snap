// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maintenance Tab Container renders the maintenance tab container correctly 1`] = `
Array [
  <div
    className="sc-bBXqnf cKDpGo"
  >
    <div
      className="sc-fodVxV BISWi"
    >
      <div
        className="left"
      >
        SideNavigation
        <img
          alt="Protostellar globe"
          className="sc-kEjbxe qRbSs"
          src="ProtostellarGlobe.svg"
        />
        <span
          className="sc-iqHYGH fOGDer"
        >
          <a
            className="sc-bqyKva hDtbLJ"
            href="/track-and-predict"
            onClick={[Function]}
          >
            <span
              className="link"
            >
              Track&Predict
            </span>
          </a>
          <span
            className="sc-dQppl ibVCiw"
          >
            /
          </span>
          <a
            className="sc-bqyKva hDtbLJ"
            href="/"
            onClick={[Function]}
          >
            <span
              className="link"
            >
              unknown
            </span>
          </a>
        </span>
      </div>
      <nav
        className="sc-fFubgz deccwI"
      >
        <div
          className="sc-kstrdz lolpkk tpz-header white"
        >
          <a
            className="sc-hBEYos iRgagf"
            onClick={[Function]}
          >
            navbar.documentation
          </a>
          <a
            className="sc-hBEYos iRgagf"
            href="mailto:support-1logistics@thalesdigital.io"
          >
            <i
              className="material-icons"
            >
              help_outline
            </i>
          </a>
          <a
            className="sc-hBEYos iRgagf"
          >
            <div
              className="tpz-menu-options tpz-header-mobile-menu-options"
            >
              <div
                className="tpz-menu"
              >
                <button
                  className="tpz-menu-button"
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  type="button"
                >
                  <i
                    className="material-icons"
                  >
                    account_circle
                  </i>
                </button>
              </div>
            </div>
          </a>
        </div>
      </nav>
    </div>
  </div>,
  <div
    className="sc-iwyYcG kkPWSB"
  >
    <div
      className="sc-cxFLnm eaKehO"
    >
      <nav
        className="sc-pFZIQ iOsina"
      >
        <a
          className="sc-fubCfw sc-jrAGrp YVgyN hwvkzP  active-nav-item  "
          href="/track-and-predict/maintenance-asset//checklists"
          onClick={[Function]}
        >
          maintenance.tabs.checklists
        </a>
        <a
          className="sc-fubCfw sc-jrAGrp YVgyN hwvkzP   "
          href="/track-and-predict/maintenance-asset//damages"
          onClick={[Function]}
        >
          maintenance.tabs.damages
        </a>
        <a
          className="sc-fubCfw sc-jrAGrp YVgyN hwvkzP   "
          href="/track-and-predict/maintenance-asset//maintenance-parameters"
          onClick={[Function]}
        >
          maintenance.tabs.parameters
        </a>
        <a
          className="sc-fubCfw sc-jrAGrp YVgyN hwvkzP   "
          href="/track-and-predict/maintenance-asset//maintenance-log"
          onClick={[Function]}
        >
          maintenance.tabs.regular
        </a>
        <a
          className="sc-fubCfw sc-jrAGrp YVgyN hwvkzP   "
          href="/track-and-predict/maintenance-asset//activity"
          onClick={[Function]}
        >
          maintenance.tabs.activity
        </a>
        <a
          className="sc-fubCfw sc-jrAGrp YVgyN hwvkzP   "
          href="/track-and-predict/maintenance-asset//mileage"
          onClick={[Function]}
        >
          maintenance.tabs.mileage
        </a>
      </nav>
      Children
    </div>
    <div
      className="sc-lmoMRL ktQhwL"
    >
      <div
        className="sc-iBPRYJ ulSNt"
      >
        <div
          className="sc-bdfBwQ bQlKor"
        >
          <div
            className="sc-gsTCUz gSBHTq"
          >
            <div
              className="sc-dlfnbm jFixhj"
            >
              <img
                alt="protostellar-logo"
                height="32px"
                src="Protostellar_Logo_grau_2-02.svg"
              />
            </div>
            <div
              className="sc-hKgILt edhEfs"
            >
              <ul
                className="sc-eCssSg cUiBZU"
              >
                <li
                  className="sc-jSgupP fsVOKa"
                >
                  <a
                    className="sc-gKsewC hFWEic"
                    onClick={[Function]}
                  >
                    footer.notice
                  </a>
                </li>
                <li
                  className="sc-jSgupP fsVOKa"
                >
                  Protostellar
                   
                  0.0.1
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
